lib/segment/src/payload_storage/query_checker.rs

```diff
@@ -264,8 +272,44 @@ mod tests {
             },
         }));
 
+        let is_empty_condition_3 = Filter::new_must(Condition::IsEmpty(IsEmptyCondition {
+            is_empty: PayloadField {
+                key: "parts".to_string(),
+            },
+        }));
+
         assert!(!payload_checker.check(0, &is_empty_condition_1));
         assert!(payload_checker.check(0, &is_empty_condition_2));
+        assert!(payload_checker.check(0, &is_empty_condition_3));
+
+        let is_null_condition_1 = Filter::new_must(Condition::IsNull(IsNullCondition {
```

Comment:
Minor nitpick: you can probably just assert directly to reuse the same variable without a counter.

```rust
let is_null_condition = Filter::new_must(Condition::IsNull(IsNullCondition {
    is_null: PayloadField {
        key: "amount".to_string(),
    },
}));
assert!(!payload_checker.check(0, &is_null_condition));
```

Reply:
And it will be more readable with each assertion after the assignment. Great idea, thanks.